Makefile

```diff
@@ -21,21 +22,38 @@ endif
 all:
 	$(MSBUILD)
 
-prepare::
+prepare:: prepare-props
 	git submodule update --init --recursive
 	nuget restore
 	(cd external/Java.Interop && nuget restore)
+
+prepare-props:
 	cp Configuration.Java.Interop.Override.props external/Java.Interop/Configuration.Override.props
+	./build-tools/scripts/generate-os-info Configuration.OperatingSystem.props
 	cp `$(MSBUILD) /nologo /v:minimal /t:GetMonoSourceFullPath build-tools/scripts/Paths.targets`/mcs/class/msfinal.pub .
 
 include build-tools/scripts/BuildEverything.mk
 
+# Please keep the package names sorted
 ifeq ($(OS),Linux)
-UBUNTU_DEPS          = libzip4 curl openjdk-8-jdk git make automake autoconf libtool unzip vim-common clang lib32stdc++6 lib32z1
+UBUNTU_DEPS          = autoconf autotools-dev automake clang curl g++-mingw-w64 gcc-mingw-w64 git libtool libzip4 linux-libc-dev make openjdk-8-jdk unzip vim-common
```

Comment:
This is now at the point where I'd suggest a multi-line definition:

    UBUNTU_DEPS = \
        autoconf \
        autotools-dev \
        ...

It'll make for smaller changes as future values are added.